onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -477,7 +511,7 @@ std::shared_ptr<IExecutionProviderFactory> DMLProviderFactoryCreator::CreateFrom
 // The OrtSessionOptionsAppendExecutionProvider_DML export on the OrtDmlApi should be used instead.
 ORT_API_STATUS_IMPL(OrtSessionOptionsAppendExecutionProvider_DML, _In_ OrtSessionOptions* options, int device_id) {
 API_IMPL_BEGIN
-  options->provider_factories.push_back(onnxruntime::DMLProviderFactoryCreator::Create(device_id));
+  options->provider_factories.push_back(onnxruntime::DMLProviderFactoryCreator::Create(device_id, false, false, false));
```

Comment:
It looks like we are allowing user to enable `enable_dynamic_graph_fusion` only by [OrtApis::SessionOptionsAppendExecutionProvider](https://github.com/microsoft/onnxruntime/blob/35ecce45496ee752bc3f85618eba713bb50e6069/onnxruntime/core/session/provider_registration.cc#L76C43-L76C49) api option.
[Should we also them to pass it by OrtDmlApi::SessionOptionsAppendExecutionProvider_DML](https://github.com/microsoft/onnxruntime/blob/35ecce45496ee752bc3f85618eba713bb50e6069/onnxruntime/core/providers/dml/dml_provider_factory.cc#L546)?

Reply:
I'd rather limit the scope for now since it's not clear how useful it will be in the long term, or if the API will need to change. Each time we add new options to `OrtSessionOptionsAppendExecutionProvider_DML` we need to add a new version of the function while keeping the old ones there. We can add it in future releases if there's a real need to use it from the C++ side.